### PR TITLE
Remove and search for packages outdated from recipe (#1394)

### PR DESCRIPTION
**Pull request by et666**
I added the functionality to search for or remove outdated packages.

Normally all packages should have the same recipe hash. But to make it more fail safe the method get_recipe_hashs() looks for each package for the right recipe hash.

### CODE REVIEW

**Comment by et666** [comment]
Good to have tests...
I am looking into the problem

**Comment by memsharded** [comment]
Thanks very much for contributing this. We will also have a look when possible, to check what it could be failing. Maybe something related to the local/remote. As the search should work also remotely, not only locally, probably the way of getting the recipe hash from the local storage might be breaking for server queries. Not sure about how to approach it, so don't worry about it, we will help. Thanks again!

**Comment by memsharded** [comment]
Will try to make it into 0.25, if we see it is more complex than expected, we will delay until 0.26

**Comment by memsharded** [comment]
It turns out that this is more complex than expected. I have some ongoing work, but still have to put a couple of hours more in it. Will do a PR to yours when ready.

**Comment by memsharded** [comment]
Proposed improvements in https://github.com/et666/conan/pull/1

File: conans/test/command/search_test.py [hunk 1]
```diff
@@ -244,7 +244,7 @@ def _assert_pkg_q(self, query, packages_found, remote):
         if remote:
             command += " -r %s" % remote
         self.client.run(command, ignore_error=True)
-        print self.client.user_io.out
+        print(self.client.user_io.out)
```
Comment (memsharded) [inline conans/test/command/search_test.py]: Actually this was  my mistake, I left a debugging print there 😅 
Together with the ``ignore_error=True`` could be removed.

**Comment by memsharded** [comment]
As I have collaborated in the PR, assigned @lasote to review it.

**Pull request merged**